src/backend/cuda/cast.hpp

```diff
@@ -23,18 +23,86 @@ namespace cuda
 template<typename To, typename Ti>
 struct CastOp
 {
-    std::string func;
-    CastOp() {
-        std::string tmp = std::string("___mk") + afShortName<To>();
-        func = cuMangledName<Ti, false>(tmp.c_str());
+    const char *name()
+    {
+        return "";
+    }
+};
+
+#define CAST_FN(TYPE)                           \
+    template<typename Ti>                       \
+        struct CastOp<TYPE, Ti>                 \
+    {                                           \
+        const char *name()                      \
+        {                                       \
+            return "("#TYPE")";                 \
+        }                                       \
+    };
+
+CAST_FN(int)
+CAST_FN(unsigned int)
+CAST_FN(unsigned char)
+CAST_FN(unsigned short)
+CAST_FN(short)
+CAST_FN(float)
+CAST_FN(double)
+
+#define CAST_CFN(TYPE)                          \
+    template<typename Ti>                       \
+    struct CastOp<TYPE, Ti>                     \
+    {                                           \
+        const char *name()                      \
+        {                                       \
+            return "__convert_"#TYPE;           \
+        }                                       \
+    };
+
+
+CAST_CFN(cfloat)
+CAST_CFN(cdouble)
+CAST_CFN(char)
+
+
+template<>
+struct CastOp<cfloat, cdouble>
+{
+    const char *name()
+    {
+        return "__convert_z2c";
+    }
+};
+
+
```

Comment:
extra newline here and L65 and L60
(Nitpicking, but since the code looks so good and you already edit the line, it deserves to be perfect... ;-))